src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -314,17 +314,9 @@ private static void ApplyDotNetSdkEnvironmentVariables(string dotNetSdkPath)
             const string MSBuildExtensionsPath = nameof(MSBuildExtensionsPath);
             const string MSBuildSDKsPath = nameof(MSBuildSDKsPath);
 
-            var variables = new Dictionary<string, string>
-            {
-                [MSBUILD_EXE_PATH] = dotNetSdkPath + "MSBuild.dll",
-                [MSBuildExtensionsPath] = dotNetSdkPath,
-                [MSBuildSDKsPath] = dotNetSdkPath + "Sdks"
-            };
-
-            foreach (var kvp in variables)
-            {
-                Environment.SetEnvironmentVariable(kvp.Key, kvp.Value);
-            }
+            Environment.SetEnvironmentVariable(MSBUILD_EXE_PATH, Path.Combine(dotNetSdkPath, "MSBuild.dll"));
+            Environment.SetEnvironmentVariable(MSBuildExtensionsPath, dotNetSdkPath);
+            Environment.SetEnvironmentVariable(MSBuildSDKsPath, Path.Combine(dotNetSdkPath, "Sdks"));
```

Comment:
I left this comment on the issue, but I'd like to suggest adding sanity checks to ensure that these paths all exist just in case.